crates/astria-sequencer-relayer/src/sequencer_relayer.rs

```diff
@@ -83,22 +79,19 @@ impl SequencerRelayer {
             format!("failed to parse provided `api_addr` string as socket address: `{api_addr}`",)
         })?;
         let api_server = api::start(api_socket_addr, state_rx);
-        Ok(Self {
+        let relayer = Self {
             api_server,
             relayer,
-            shutdown_token,
-        })
+            shutdown_token: shutdown_handle.token(),
+        };
+        Ok((relayer, shutdown_handle))
     }
 
     pub fn local_addr(&self) -> SocketAddr {
         self.api_server.local_addr()
     }
 
-    /// Run Sequencer Relayer.
-    ///
-    /// # Panics
-    /// Panics if a signal listener could not be constructed (usually if this binary is not run on a
-    /// Unix).
+    /// Runs Sequencer Relayer.
     pub async fn run(self) {
```

Comment:
I wonder if we should just go with what hyper and others are doing and just call this `spawn`, and internally `tokio::spawn` the process, returning a `Handle` to the object (which itself contains the `JoinHandle` and the token).

Reply:
As discussed elsewhere, I think it's probably worth keeping the creation of the token to the `new` function so we have the option to have shutdown handling already in place before we actually start running the relayer.